tests/test_debian_9.py

```diff
@@ -814,38 +816,121 @@ def test_compose_2idx2sh(self):
         self.project_name = generate_random_string()
         container_count, rc = self.compose_up()
 
-        output_sh1 = self.get_container_logs("sh1")
-        output_sh2 = self.get_container_logs("sh2")
-        output_idx1 = self.get_container_logs("idx1")
-        output_idx2 = self.get_container_logs("idx2")
-        log_json_sh1 = self.extract_json("sh1")
-        log_json_sh2 = self.extract_json("sh2")
-        log_json_idx1 = self.extract_json("idx1")
-        log_json_idx2 = self.extract_json("idx2")
+        container_names = ["idx1", "idx2", "sh1", "sh2"]
+        # Check ansible version & configs
+        logs_output = [self.get_container_logs(x) for x in container_names]
+        for log in logs_output:
+            self.check_ansible(log)
         assert rc == 0
         # Wait for containers to be healthy
         assert self.wait_for_containers(container_count)
         # Check Splunkd on all the containers
         assert self.check_splunkd("admin", self.password)
+        # Check values in log output
+        log_json = {}
+        for c in container_names:
+            log_json[c] = self.extract_json(c)
+        self.check_common_keys(log_json["sh1"], "sh")
+        self.check_common_keys(log_json["sh2"], "sh")
+        self.check_common_keys(log_json["idx1"], "idx")
+        self.check_common_keys(log_json["idx2"], "idx")
+        try:
+            for c in container_names:
+                assert log_json[c]["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
+                assert log_json[c]["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
+        except KeyError as e:
+            self.logger.error(e)
+            assert False
+
+        # Check connections
+        idx_list = ["idx1", "idx2"]
+        containers = self.client.containers(filters={"label": "com.docker.compose.project={}".format(self.project_name)})
+        
+        for container in containers:
+            c_name = container["Labels"]["com.docker.compose.service"]
+            if c_name == "sh1" or c_name == "sh2":
+                splunkd_port = self.client.port(container["Id"], 8089)[0]["HostPort"]
+                status, content = self.handle_request_retry("GET", "https://localhost:{}/services/search/distributed/peers?output_mode=json".format(splunkd_port), 
+                                                    {"auth": ("admin", self.password), "verify": False})
+                assert status == 200
+                output = json.loads(content)
+                peers = [x["content"]["peerName"] for x in output["entry"]]
+                assert len(peers) == 2 and set(peers) == set(idx_list)
+
+    def test_compose_2idx2sh1cm(self):
```

Comment:
with these and all the other tests, we should be documenting what the tests' goal is.  Long term in engineering, you'll find tests that grow stale, due to the way things are done, and underlying code being changed.  By writing the test's purpose, we can make sure that an old test is replaced with new methods, and still hits the original feature goal.  Not saying you need to add the doc strings, just keep that for your back pocket.